File: src/main/java/kitchenpos/common/DomainService.java
package kitchenpos.common;

import org.springframework.stereotype.Service;

import java.lang.annotation.*;

/**
 * 도메인 서비스를 나타내는 어노테이션
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Service
public @interface DomainService {
}

File: src/main/kotlin/kitchenpos/common/annotation/DomainService.kt
package kitchenpos.common.annotation

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Service


@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Service
annotation class DomainService(
    @get:AliasFor(annotation = Service::class)
    val value: String = ""
)
